core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -461,19 +464,28 @@ default ScheduledExecutorService makeContextAware(ScheduledExecutorService execu
     <T extends Future<?>> GenericFutureListener<T> makeContextAware(GenericFutureListener<T> listener);
 
     /**
-     * Returns a {@link CompletionStage} that makes sure the current {@link CompletionStage} is set and
+     * Returns a {@link CompletionStage} that makes sure the current {@link RequestContext} is set and
      * then invokes the input {@code stage}.
      */
     <T> CompletionStage<T> makeContextAware(CompletionStage<T> stage);
 
     /**
-     * Returns a {@link CompletableFuture} that makes sure the current {@link CompletableFuture} is set and
+     * Returns a {@link CompletableFuture} that makes sure the current {@link RequestContext} is set and
      * then invokes the input {@code future}.
      */
     default <T> CompletableFuture<T> makeContextAware(CompletableFuture<T> future) {
         return makeContextAware((CompletionStage<T>) future).toCompletableFuture();
     }
 
+    /**
+     * Returns a {@link Logger} which prepends this {@link RequestContext} to the log message.
+     *
+     * @param logger the {@link Logger} to decorate.
+     */
+    default Logger makeContextAware(Logger logger) {
```

Comment:
Question: I'm wondering if we still need this methods because we add the `ctx` in `LoggingDecorator`. 🤔

Reply:
This method is for a user who wants to make any logger context-aware. It's used for logging application-level logs.